app/models/post.rb

```diff
@@ -0,0 +1,17 @@
+class Post < ApplicationRecord
+  include PgSearch::Model
+
+  pg_search_scope :search_by_title, against: :title
+
+  validates :author, presence: true
+  validates :hn_created_at, presence: true
+  validates :hn_id, presence: true, uniqueness: { case_sensitive: false }
+  validates :post_type, presence: true
+  validates :title, presence: true
+  validates :url, presence: true
+
+  has_many :stars, dependent: :destroy
+  has_many :starred_users, class_name: 'User', foreign_key: 'user_id', through: :stars, source: :user
```

Comment:
The use of `starred_users` vs plain `users` was to avoid confusion. It would be reasonable to see `Post.users` and believe that `User` may be the author or creator of the `Post`. `starred_users` is a little more explicit to the context of it - the `User` that `Star` (starred) the post.